backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -796,12 +821,58 @@
     /// <summary>
     /// must be called as part of an lcm action
     /// </summary>
-    internal void AddComplexFormComponent(ILexEntry lexEntry, ComplexFormComponent component)
+    internal void AddComplexFormComponent(ILexEntry lexComplexForm, ComplexFormComponent component, BetweenPosition<ComplexFormComponent>? between = null)
     {
         ICmObject lexComponent = component.ComponentSenseId is not null
             ? SenseRepository.GetObject(component.ComponentSenseId.Value)
             : EntriesRepository.GetObject(component.ComponentEntryId);
-        lexEntry.AddComponent(lexComponent);
+        InsertComplexFormComponent(lexComplexForm, lexComponent, between);
+    }
+
+    internal void InsertComplexFormComponent(ILexEntry lexComplexForm, ICmObject lexComponent, BetweenPosition<ComplexFormComponent>? between = null)
+    {
+        var previousComponentId = between?.Previous?.ComponentSenseId ?? between?.Previous?.ComponentEntryId;
+        var nextComponentId = between?.Next?.ComponentSenseId ?? between?.Next?.ComponentEntryId;
+
+        var entryRef = lexComplexForm.ComplexFormEntryRefs.SingleOrDefault();
```

Comment:
should we just use `FirstOrDefault` since you've seen cases where there's more than one?

Reply:
This case probably doesn't matter too much. If we got this far, then we know there's only one.
I think I'll just leave it until we actually decide how to handle this.